src/main/java/org/opentripplanner/routing/trippattern/TripTimes.java

```diff
@@ -365,20 +365,29 @@ public void setRealTimeState(final RealTimeState realTimeState) {
      */
     public boolean timesIncreasing() {
         final int nStops = scheduledArrivalTimes.length;
-        int prevDep = -1;
+        int prevDep = Integer.MIN_VALUE;
+        int prevIndex = -1;
         for (int s = 0; s < nStops; s++) {
+            // Do not validate for cancelled stops
+            if (isCancelledStop(s)) {
+                continue;
+            }
+            boolean isFirstStop = s == 0;
+            boolean isLastStop = s + 1 == nStops;
             final int arr = getArrivalTime(s);
             final int dep = getDepartureTime(s);
 
-            if (dep < arr) {
+            // Do not validate for first and last stop
+            if (!isFirstStop && !isLastStop && dep < arr) {
```

Comment:
OTP algorithm expects in some places, eg. [here](https://github.com/opentripplanner/OpenTripPlanner/blob/86a118858f48c152de7d7732c44a25fd27987072/src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/request/TripScheduleWithOffset.java#L29), that even the first and last stop time have a non-negative dwell time. It can be zero if needed.

Reply:
We should discuss this in the next OTP meeting. What purpose does these times play? I think at least Raptor and OptimizedTransfers ignore these. If we can I think also the rest of the system should. We need the first arrival-time and last-departure time for board at sops and such, so we should establish some business rules for this.

Reply:
I've had some discussion about it with @lassetyr. Possible solution for this problem would be to make sure Arrival and Departure times are set on all stops. In case time is not present in realtime message we could just populate it with time from departure/arrival.